pkg/loader/compose/v1v2.go

```diff
@@ -268,6 +269,51 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
 			log.Infof("Service name in docker-compose has been changed from %q to %q", name, normalizeServiceNames(name))
 		}
 	}
+
+	//Handling Volumes
+	for name, _ := range komposeObject.ServiceConfigs {
+		vols := handleVolume(name, komposeObject)
+		var FinalVol []kobject.Volumes
+		for _, v := range vols {
+			if !findDuplicateVol(v, FinalVol) {
+				FinalVol = append(FinalVol, v)
+			}
+		}
+		var count int
+		for v, _ := range FinalVol {
+			vol := &FinalVol[v]
+			var err error
+			vol.VolumeName, vol.Host, vol.Container, vol.Mode, err = transformer.ParseVolume(vol.MountPath)
+			if err != nil {
+				log.Warningf("Failed to configure container volume: %v", err)
+				continue
+			}
+			log.Debug("Volume name %s", vol.VolumeName)
+
+			if vol.VFrom == "" {
+				vol.PVCName = fmt.Sprintf("%s-claim%d", vol.SvcName, count)
+				count++
+			}
+
+		}
+		var temp = komposeObject.ServiceConfigs[name]
+		temp.Volumes = FinalVol
+		komposeObject.ServiceConfigs[name] = temp
+	}
```

Comment:
There's no way we can add this much code into this fuction, could you refactor this into a separate function (see my other comment too)